src/components/section/SectionFilter/SectionFilter.tsx

```diff
@@ -0,0 +1,109 @@
+import classNames from 'classnames'
+import { Form, Formik, useFormikContext } from 'formik'
+import {
+  ReactChildren,
+  ReactElement,
+  ReactNode,
+  useEffect,
+  useState,
+} from 'react'
+import { useTheme } from '../../../framework'
+import { ClassNameProps } from '../../types'
+import { SubmitButton } from '../../button/SubmitButton'
+import { FilterDialog } from './FilterDialog'
+
+type Props<T> = ClassNameProps & {
+  initialValues: T
+  onFilter: (values: T) => void
+  onCloseFilter: () => void
+  showFilterPopupBelowBreakpoint?: number
+  confirmationButtonContent: ReactNode
+  children: ReactChildren
+}
+
+function SubmitOnChange(): null {
+  const formik = useFormikContext()
+
+  useEffect(() => {
+    if (formik.isValid && formik.dirty) {
+      formik.submitForm()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [formik.isValid, formik.dirty, formik.values])
+
+  return null
+}
+
+function useBelowBreakpoint(breakpoint: number): boolean {
+  const isWindowClient = typeof window === 'object'
+
+  const [windowSize, setWindowSize] = useState<number | undefined>(
+    isWindowClient ? window.innerWidth : undefined
+  )
+
+  useEffect(() => {
+    function setSize() {
+      setWindowSize(window.innerWidth)
+    }
+
+    if (isWindowClient) {
+      window.addEventListener('resize', setSize)
+
+      return () => window.removeEventListener('resize', setSize)
+    }
+  }, [isWindowClient, setWindowSize])
+
+  return !!windowSize && breakpoint > windowSize
+}
+
+export function SectionFilter<T>({
+  className,
+  initialValues,
+  onFilter,
+  onCloseFilter,
+  showFilterPopupBelowBreakpoint = 768,
```

Comment:
What do you think about these names, to avoid reusing the word filter too much?

```suggestion
  onClose,
  showPopupBelowBreakpoint = 768,
```